src/main/java/randoop/condition/SpecificationCollection.java

```diff
@@ -0,0 +1,312 @@
+import randoop.condition.specification.OperationSignature;
+import randoop.condition.specification.OperationSpecification;
+import randoop.reflection.TypeNames;
+import randoop.util.Log;
+import randoop.util.MultiMap;
+
+/**
+ * A collection of {@link OperationSpecification} objects. Also a map from an {@link
+ * AccessibleObject} reflection object to the {@link OperationConditions} for the corresponding
+ * {@link randoop.operation.TypedClassOperation}.
+ *
+ * <p>The {@link SpecificationCollection} should be constructed from the specification input before
+ * the {@link randoop.reflection.OperationModel} is created.
+ *
+ * <p>This class stores the {@link OperationSpecification} objects, and only constructs the
+ * corresponding {@link OperationConditions} on demand. This lazy-strategy avoids building
+ * conditions methods for specifications that are not used.
+ */
+public class SpecificationCollection {
+
+  /** The map from reflection objects to the corresponding {@link OperationSpecification} */
+  private final Map<AccessibleObject, OperationSpecification> specificationMap;
+
+  /** The map from method signatures to methods with that signature and specifications */
+  private final MultiMap<OperationSignature, Method> signatureMap;
+
+  /** The map from reflection object to overridden method with specification */
+  private final Map<AccessibleObject, Set<Method>> parentMap;
+
+  /** The compiler for creating conditionMethods */
+  private final SequenceCompiler compiler;
+
+  /** Map for memoizing conditions for specifications converted by parent search */
+  private Map<AccessibleObject, OperationConditions> conditionMap;
+
+  /**
+   * Creates a {@link SpecificationCollection} for the given specification map.
+   *
+   * <p>This constructor is used internally. It is only accessible to allow testing. Clients should
+   * use {@link #create(List)} instead.
+   *
+   * @param specificationMap the map from reflection objects to {@link OperationSpecification}
+   * @param signatureMap the multimap from a signature to methods with with the signature
+   * @param parentMap the map from a method to methods that it it overrides and that have a
+   *     specification
+   */
+  SpecificationCollection(
+      Map<AccessibleObject, OperationSpecification> specificationMap,
+      MultiMap<OperationSignature, Method> signatureMap,
+      Map<AccessibleObject, Set<Method>> parentMap) {
+    this.specificationMap = specificationMap;
+    this.signatureMap = signatureMap;
+    this.parentMap = parentMap;
+    this.conditionMap = new HashMap<>();
+    SequenceClassLoader sequenceClassLoader = new SequenceClassLoader(getClass().getClassLoader());
+    List<String> options = new ArrayList<>();
+    this.compiler = new SequenceCompiler(sequenceClassLoader, options);
+  }
+
+  /**
+   * Creates a {@link SpecificationCollection} from the list of files of serialized specifications.
+   *
+   * @param specificationFiles the files of serialized specifications
```

Comment:
What format do these have?  How were they created?  What do they represent (eg, all specifications from an entire project, or less than that)?